src/data/padics/ring_homs.lean

```diff
@@ -0,0 +1,689 @@
+/-
+Copyright (c) 2020 Johan Commelin and Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin and Robert Y. Lewis
+-/
+
+import data.padics.padic_integers
+
+/-!
+
+# Relating `ℤ_[p]` to `zmod (p ^ n)`
+
+In this file we establish connections between the `p`-adic integers $\mathbb{Z}_p$
+and the integers modulo powers of `p`, $\mathbb{Z}/p^n\mathbb{Z}$.
+
+## Main declarations
+
+We show that $\mathbb{Z}_p$ has a ring hom to $\mathbb{Z}/p^n\mathbb{Z}$ for each `n`.
+The case for `n = 1` is handled separately, since it is used in the general construction
+and we may want to use it without the `^1` getting in the way.
+* `padic_int.to_zmod`: ring hom to `zmod p`
+* `padic_int.to_zmod_pow`: ring hom to `zmod (p^n)`
+* `padic_int.ker_to_zmod` / `padic_int.ker_to_zmod_pow`: the kernels of these maps are the ideals
+  generated by `p^n`
+
+We also establish the universal property of $\mathbb{Z}_p$ as a projective limit.
+Given a family of compatible ring homs $f_k : R \to \mathbb{Z}/p^n\mathbb{Z}$,
+there is a unique limit $R \to \mathbb{Z}_p$.
```

Comment:
I think LaTeX in docstrings is cool but does it get rendered nicely anywhere?

Reply:
Yes, it gets rendered in the docs. Unfortunately I couldn't find an example quickly. I thought it got rendered in VSCode tooltips as well, but it looks like not. That's not so relevant for the module doc. 

I can change it of course, but the unicode version of this stuff is ugly. Do we write the p-adic integers as `ℤ_[p]` (Lean syntax) or `ℤ_p`? `ℤ/p^nℤ` isn't nice.

Reply:
Here's an example, malformed right now but fixed by #3965 : https://leanprover-community.github.io/mathlib_docs/representation_theory/maschke.html